src/bun.js/bindings/webcore/WebSocket.cpp

```diff
@@ -425,13 +426,77 @@ ExceptionOr<void> WebSocket::connect(const String& url, const Vector<String>& pr
     if (!protocols.isEmpty())
         protocolString = joinStrings(protocols, subprotocolSeparator());
 
-    ZigString host = Zig::toZigString(m_url.host());
-    auto resource = resourceName(m_url);
-    ZigString path = Zig::toZigString(resource);
     ZigString clientProtocolString = Zig::toZigString(protocolString);
-    uint16_t port = is_secure ? 443 : 80;
-    if (auto userPort = m_url.port()) {
-        port = userPort.value();
+    ZigString host;
+    ZigString path;
+    uint16_t port;
+
+    String requestPath;
+    String resource;
+
+    if (is_unix) {
+        // ws+unix URL format: ws+unix://socket_path:request_path
+        // https://github.com/websockets/ws/blob/master/doc/ws.md#ipc-connections
+        // Windows: ws+unix://C:/path/to/pipe:/pathname (gets parsed as host="C:", path="/path/to/pipe:/...")
+        // Unix: ws+unix:///absolute/path/to/socket:/pathname?search_params
+        StringView urlPath = m_url.path();
+        size_t colonPos = notFound;
+
+#if OS(WINDOWS)
+        // On Windows, the URL parser splits C:/path into host="C:" and path="/path"
+        // We need to reconstruct the full Windows path with backslashes
+        auto urlHost = m_url.host();
+        String normalizedPath;
+        if (!urlHost.isEmpty() && urlHost.length() <= 2) {
+            // Likely a Windows drive letter (e.g., "C:")
+            // Reconstruct full path: C: + /Windows/Temp/... -> C:\Windows\Temp\...
+            normalizedPath = makeString(urlHost, urlPath);
+            normalizedPath.replace('/', '\\');
+            urlPath = normalizedPath;
+        } else if (urlPath.startsWith("//."_s)) {
+            // Windows named pipe path like //./pipe/name
+            normalizedPath = urlPath.toString();
+            normalizedPath.replace('/', '\\');
+            urlPath = normalizedPath;
+        }
+
+        // Skip drive letter colon if present (e.g., "C:\path\socket:/" -> find second colon)
+        if (urlPath.length() >= 2 && urlPath[1] == ':') {
+            // Has drive letter, search for colon after position 2
+            colonPos = urlPath.find(':', 2);
+        } else {
+            colonPos = urlPath.find(':');
+        }
+#else
+        colonPos = urlPath.find(':');
+#endif
+
+        if (colonPos == notFound) {
+            host = Zig::toZigString(urlPath);
+            requestPath = "/"_s;
+        } else {
+            host = Zig::toZigString(urlPath.substring(0, colonPos));
+            requestPath = urlPath.substring(colonPos + 1).toString();
+            if (requestPath.isEmpty())
+                requestPath = "/"_s;
+        }
+        if (!requestPath.startsWith('/'))
+            requestPath = makeString("/"_s, requestPath);
+
+        auto query = m_url.queryWithLeadingQuestionMark();
+        if (!query.isEmpty())
+            requestPath = makeString(requestPath, query);
+
+        path = Zig::toZigString(requestPath);
```

Comment:
This should use WTF::URL's fileSystemPath instead of manual normalization logic I think?

For example:

https://github.com/oven-sh/bun/blob/da53deae80c4618d00cb96a293a46cc9785c5988/src/bun.js/bindings/ImportMetaObject.cpp#L705-L713